lib/NetPacksLib.cpp

```diff
@@ -240,6 +240,15 @@ DLL_LINKAGE void SetAvailableHeroes::applyGs( CGameState *gs )
 	for (int i = 0; i < GameConstants::AVAILABLE_HEROES_PER_PLAYER; i++)
 	{
 		CGHeroInstance *h = (hid[i]>=0 ?  gs->hpool.heroesPool[hid[i]].get() : nullptr);
+		for (auto I = h->artifactsInBackpack.begin(), E = h->artifactsInBackpack.end();
+		     I != E; ++I)
+		{
+			if (I->artifact->artType->id == ArtifactID::GRAIL)
+			{
+				h->artifactsInBackpack.erase(I);
+				break;

```

Comment:
There is no "erase_if" but erase + remove_if. And there will never be more than one grail (I saw the code, it would be 100% rewritten to allow multiple instances). Personally, I forbid such STL usage in Samsung production-close projects because of subsequent maintenance cost and don't recommend to use it here.